src/hooks/useAlternateLinks.ts

```diff
@@ -0,0 +1,41 @@
+import { useRouter } from 'next/router';
+import { useEffect, useState } from 'react';
+import { getOrigin } from '../helpers/getOrigin';
+import { getSourceUrl } from '../helpers/getSourceUrl';
+import { useRewrites } from '../hooks/useRewrites';
+
+interface AlterNateLinks {
+  href: string;
+  hrefLang: string;
+}
+
+export const useAlternateLinks = (currentLocale: string): AlterNateLinks[] | null => {
+  const { basePath, pathname, locales } = useRouter();
+  const rewrites = useRewrites();
+  const [alternateLinks, setAlternateLinks] = useState<AlterNateLinks[]>([]);
+  const path = `${basePath ?? ''}${pathname}`;
+
+  useEffect(() => {
+    const origin = getOrigin();
+
+    /*
+      If we are at the root '/' of the website we should build alternate
+      links for all the locales
+    */
+    const alternateLocales =
+      pathname === '/' ? locales : locales.filter((lang) => lang !== currentLocale);
```

Comment:
```suggestion
    const alternateLocales = locales;
```

The alternate links are expected to always include also the current page. I've found this to be a pretty good reference for these: https://developers.google.com/search/docs/advanced/crawling/localized-versions